src/App.tsx

```diff
@@ -1,27 +1,82 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Loader from 'react-loaders';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, getFirst5, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const fetchData = async (getData: { (): Promise<Good[]>; }) => {
+    setIsLoading(true);
+    setVisibleGoods([]);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    try {
+      const data = await getData();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      setVisibleGoods(data);
+    } catch (error) {
+      setHasLoadingError(true);
+      throw new Error('No data was found');
+    } finally {
+      setIsLoading(false);
+    }
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <div className="App__buttons">
+        <button
+          type="button"
+          data-cy="all-button"
+          onClick={async () => {
+            await fetchData(getAll);
+          }}
```

Comment:
```suggestion
          onClick={() => fetchData(getAll)}
```

You don't need to wait for it here. It blocks UI